File: go/12_file_io.go
package main

import (
	"fmt"
	"os"
)

func main() {
	err := os.WriteFile("test.txt", []byte("hello from go\nsecond line\n"), 0644)
	if err != nil {
		fmt.Println(err)
		return
	}

	data, err := os.ReadFile("test.txt")
	if err != nil {
		fmt.Println(err)
		return
	}
	fmt.Print(string(data))
}

File: javascript/12_file_io.js
// 12 file i/o
const fs = require("fs");

fs.writeFileSync("test.txt", "hello from javascript\nsecond line\n");

const content = fs.readFileSync("test.txt", "utf8");
console.log(content);

const lines = content.split("\n");
lines.forEach(line => {
  if (line) console.log("line:", line);
});
